### PR TITLE
feature(Project): Improved submission system

### LINKED ISSUE #56: Infinite scroll for submissions
Or some kind of other better system
TODO:
- [x] Update to new DB in the user project view
- [ ] Create seperate tab for submissions in user project view
- [ ] Create seperate tab for submissions in dashboard view
- [x] Create API routes for submissions:
  - [x] GET
  - [x] PATCH (update)
  - [x] DELETE
  - [x] ~~POST (create)~~ (Combined with PATCH)

### PR DESCRIPTION
**Pull request by GoldElysium**
TODO moved to #56

### CODE REVIEW

**Comment by GoldElysium** [comment]
@koleare Use this branch to work on your stuff

File: pages/projects/[id].tsx [hunk 1]
```diff
@@ -47,45 +59,51 @@ export default function ProjectPage() {
...
+						{author}
+						<h6 className="text-xl top-0 right-0 w-1/2 text-right">{`#${index+1}`}</h6>
+					</div>
+					<div className="w-full mt-3">
+						<SubmissionItem {...submission} />
+						<hr/>
+					</div>
+				</div>
+			);
+		})
```
Comment (GoldElysium) [inline pages/projects/[id].tsx:98]:
```suggestion
		});

```

File: pages/projects/[id].tsx [hunk 2]
```diff
@@ -47,45 +59,51 @@ export default function ProjectPage() {
...
+
+		return <div className="w-full h-full flex justify-center">
+			<div className="sm:w-8/12 h-full">
+				{submissionElements}
+			</div>
+		</div>;
```
Comment (GoldElysium) [inline pages/projects/[id].tsx:104]:
```suggestion
		return (
		    <div className="w-full h-full flex justify-center">
			    <div className="sm:w-8/12 h-full">
				    {submissionElements}
			    </div>
		    </div>
		);

```

File: pages/projects/[id].tsx [hunk 3]
```diff
@@ -128,26 +146,12 @@ export default function ProjectPage() {
...
-										}}
-									/>
-									{currentSubmissionIndex + 1}/{doc.submissions?.length ?? 0}
-									<ChevronRightIcon
-										className={currentSubmissionIndex + 1 < (doc.submissions?.length ?? 0) ? 'text-black h-8 w-8 cursor-pointer' : 'text-red-300 h-8 w-8'}
-										onClick={() => {
-											if (currentSubmissionIndex + 1 < (doc.submissions?.length ?? 0)) setCurrentSubmissionIndex(currentSubmissionIndex + 1);
-										}}
-									/>
+								<div className="w-full max-h-96 max-h-[960px] overflow-auto">
```
Comment (GoldElysium) [inline pages/projects/[id].tsx:153]: Why are there two `max-h`?

**Review by JackCrescend — APPROVED** [review]

**Pull request merged**